### DISCUSSION TITLE
What is the purpose of having the check for lastPressHandleTime?

### DISCUSSION THREAD

**Discussion by Freeze7x** [ppy/osu#29381] [Q&A]
https://github.com/ppy/osu/blob/dbd2e2e91f9357602ec7bc3db2410404c746de3e/osu.Game.Rulesets.Taiko/Objects/Drawables/DrawableHit.cs#L118

I don't understand the purpose of the check here that prevents a secondary input on the same frame. It just seems problematic for those not playing at high frame rates. Especially considering that lazer has an optional multithreaded input mode. Is there a reason it is here, or is it just for compatibility?

**Comment by peppy** [comment]
See the comment directly below:

https://github.com/ppy/osu/blob/dbd2e2e91f9357602ec7bc3db2410404c746de3e/osu.Game.Rulesets.Taiko/Objects/Drawables/DrawableHit.cs#L130-L131

Although arguably this isn't enough to maintain proposed behaviour because it only account for same frame.

**Reply by Freeze7x** [reply]
In somewhat extreme cases, this check can ruin doubles (1/8, 1/6, ect.) since this would eat an input unnecessarily when it thinks we are trying to hit a strong. I also can't think of a reason where this check would be beneficial to the game (anti cheese?) or the player. Would there be any issues with removing this?

**Reply by peppy** [reply]
Would need to investigate further how this is handled on stable and even Taiko no Tatsujin. I believe it is anti-cheese if anything, yes.
